src/Validator/CustomerOptionValuePriceDateOverlapConstraintValidator.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * This file is part of the Brille24 customer options plugin.
+ *
+ * (c) Brille24 GmbH
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Brille24\SyliusCustomerOptionsPlugin\Validator;
+
+use Brille24\SyliusCustomerOptionsPlugin\Entity\CustomerOptions\CustomerOptionInterface;
+use Brille24\SyliusCustomerOptionsPlugin\Entity\CustomerOptions\CustomerOptionValueInterface;
+use Brille24\SyliusCustomerOptionsPlugin\Entity\CustomerOptions\CustomerOptionValuePriceInterface;
+use Brille24\SyliusCustomerOptionsPlugin\Validator\Constraints\CustomerOptionValuePriceDateOverlapConstraint;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+class CustomerOptionValuePriceDateOverlapConstraintValidator extends ConstraintValidator
+{
+    /**
+     * Checks if the passed value is valid.
+     *
+     * @param mixed      $valuePrices The value that should be validated
+     * @param Constraint $constraint The constraint for the validation
+     */
+    public function validate($valuePrices, Constraint $constraint): void
+    {
+        Assert::isInstanceOf($valuePrices, Collection::class);
```

Comment:
Please add messages to the assertions, as debugging the generic ones can be unhelpful.